### ISSUE TITLE
Wrongly parsed program 

### ISSUE THREAD

**Issue by sancierra** [spcl/dace#332]
The following program gets parsed and validated, but does not compile. The last memlet exiting the map is wrong (third dimensions is 'z', but should be '1:O')

```

import dace
import dace.sdfg.nodes as nodes
import numpy as np

from dace.sdfg.graph import SubgraphView


N, M, O = [dace.symbol(s) for s in ['N', 'M', 'O']]
N.set(50)
M.set(60)
O.set(70)


@dace.program
def TEST(A: dace.float64[N,M,O], B: dace.float64[N,M,O], C: dace.float64[N,M,O]):
    for i, j in dace.map[0:N, 0:M]:
        with dace.tasklet:
            in1 << A[i,j,0]
            in2 << B[i,j,0]
            out >> C[i,j,0]

            out = in1 + in2

    for i,j in dace.map[0:N, 0:M]:
        for z in range(1,O):
            with dace.tasklet:
                in1 << A[i,j,z]
                in2 << B[i,j,z]
                in3 << C[i,j,0]
                out >> C[i,j,z]

                out = 2*in1 + 2*in2 + in3


def test(sdfg, graph):
    A = np.random.rand(N.get(), M.get(), O.get()).astype(np.float64)
    B = np.random.rand(N.get(), M.get(), O.get()).astype(np.float64)
    C1 = np.zeros([N.get(), M.get(), O.get()], dtype = np.float64)

    sdfg.view()
    csdfg = sdfg.compile()
    csdfg(A=A,B=B,C=C1,N=N,M=M,O=O)


if __name__ == '__main__':
    sdfg = TEST.to_sdfg()
    graph = sdfg.nodes()[0]
    test(sdfg, graph)

```

**Comment by tbennun** [comment]
I think this is already solved as part of #161 but that PR was never completed. I will take a look

**Comment by alexnick83** [comment]
This has been fixed and the fix has been pushed to master.
![nested-for-loop-fix](https://user-images.githubusercontent.com/31545860/96034863-7b1cc900-0e62-11eb-9dfe-e5013b3e7d81.png)
